subm/src/subm.js

```diff
@@ -213,6 +252,36 @@ ${Site.fileList(files)}
 `,
 });
 
+/**
+ * @param {GoogleSpreadsheetWorksheet} sheet
+ * @param {GuildMember} member
+ */
+function makeContact(sheet, member) {
+  const { user } = member;
+  if (!user) throw TypeError('user undefined');
+  return freeze({
+    member,
+    getEmail: async () => {
+      const row = await lookup(sheet, user.id);
+      if (!row) return undefined;
+      if (typeof row.email !== 'string') throw TypeError(row.email);
+      return row.email;
+    },
+
+    /**
+     * @param {string} email
+     */
+    setEmail: email =>
+      upsert(sheet, user.id, {
+        userID: user.id,
+        joined_at: member.joined_at,
+        nick: member.nick || (member.user || {}).username || '<nick???>',
```

Comment:
Save `discriminator` as well?

Reply:
the `discriminator` and all the other discord stuff gets saved in the `detail` column

Reply:
👍  totally missed that it includes the whole user object!